mesonbuild/environment.py

```diff
@@ -124,6 +124,24 @@
 
 CompilersDict = T.Dict[str, Compiler]
 
+COMPILER_OPTS_TO_ENV = {
+    'c': 'CFLAGS',
+    'cpp': 'CXXFLAGS',
+    'cuda': 'CUFLAGS',
+    'objc': 'OBJCFLAGS',
+    'objcpp': 'OBJCXXFLAGS',
+    'fortran': 'FFLAGS',
+    'd': 'DFLAGS',
+    'vala': 'VALAFLAGS',
+    'rust': 'RUSTFLAGS',
+    'pkg_config_path': 'PKG_CONFIG_PATH',
+}  # type: T.Dict[str, str]
+
+OTHER_OPTS_TO_ENV = {
```

Comment:
I'm not a big fan of "OTHER" in the name. other than what? I'm also not a big fan of each list of env dealt with duplicated but slightly different code below.

Wondering if we could consolidate all this into a single mapping:
```
ENV_TO_OPTION = {
  'CFLAGS': ['c_args'],
  ...
  'LDFLAGS': [lang + '_link_args' for lang in languages_using_ldflags]
  ...
  'PKG_CONFIG_PATH': ['pkg_config_path']
}
```
That structure can be processed in a single loop, do the value splitting if `key.endswith('_PATH')`

Reply:
Oh, you're right about the cflags/ldflags stuff for sure. In fact, using a dict is the wrong data structure, a list of tuples works much nicer.

I'm not so sure combining the _PATH variables together makes sense though. We do end up duplicated a couple of lines of code (about 4) and end up with two loops. But I think that's preferable to having an extra if path on every iteration of one loop. Look at the version I'm pushing now and see what you think.

Reply:
Yeah, `OTHER` isn't good. I changed it to `PATH_OPTS_TO_ENV`.